src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx

```
@@ -260,106 +316,204 @@ export const InfraProvisioningWizard: React.FC<InfraProvisioningWizardProps> = p
-                })
-            } else {
-              showErrorToaster((createPipelineResponse as Error)?.message)
-              setDisableBtn(false)
-              setShowPageLoader(false)
-            }
-          })
-          .catch(() => {
-            setDisableBtn(false)
-            setShowPageLoader(false)
-          })
+  const initiateAPIOperation = useCallback((loaderMessage: string): void => {
+    setDisableBtn(true)
+    setPageLoader({ show: true, message: loaderMessage })
+  }, [])
+
+  const wrapUpAPIOperation = useCallback((err?: unknown): void => {
+    if (err) {
+      showErrorToaster((err as Error)?.message)
+    }
+    setDisableBtn(false)
+    setPageLoader({ show: false })
+  }, [])
+
+  const setupPipelineWithCodebaseAndTriggers = React.useCallback((): void => {
+    if (selectRepositoryRef.current?.repository) {
+      try {
+        let setupPipelineAndTriggerPromise = getPipelineAndTriggerSetupPromise()
+        if (setupPipelineAndTriggerPromise) {
+          const { type: connectorType } = configuredGitConnector || {}
+          const { storeInGit, createBranchIfNotExists, branch } = configurePipelineRef.current
+            ?.values as SavePipelineToRemoteInterface
+          const shouldSavePipelineToGit =
+            CI_YAML_VERSIONING &&
+            connectorType &&
+            [Connectors.GITHUB, Connectors.BITBUCKET].includes(connectorType) &&
+            storeInGit
+          // if pipeline is being saved to Git and create branch if not specified is selected, we will attempt to save pipeline to Git twice
+          // Once to directly save pipeline to the specified branch with git param isNewBranch as "false"
+          // If above fails, save pipeline to the specified branch with git param isNewBranch as "true" and pass default branch of the repo as base branch
+          if (shouldSavePipelineToGit && createBranchIfNotExists) {
+            setupPipelineAndTriggerPromise.catch(_e => {
```

using `_e` here is confusing - perhaps best to give a descriptive name

might be better using try catch here aswell

---

Renamed `_e` to a better name. 

Regarding try-catch here, it's already there at the top near the beginning of the function.

---

yes but we are also using a lot of .catch() which kind of confuses the error logic

also we are calling 3 different errors `_e`, `e` and `err` which is a bit of a code smell to me

most of the error logic here ends up calling `wrapUpAPIOperation()` - which is good - but perhaps, instead of .catch() multiple times we could throw an error - then we could contain that error logic in a single catch block

---

Done, cleaned it up a little bit, this is good feedback, thanks!

---

Well I gave it my best shot, tried throwing the errors but they're swallowed somewhere in the middle and not getting caught by the outermost catch.